docroot/modules/custom/sitenow_find_text/sitenow_find_text.module

```diff
@@ -0,0 +1,198 @@
+ *
+ * @DCG
+ * This file is no longer required in Drupal 8.
+ * @see https://www.drupal.org/node/2217931
+ */
+
+/**
+ * Grab field names and mappings for all text-based fields.
+ *
+ * @return array
+ *   The field names => info mapping for all text-based fields.
+ */
+function get_all_text_fields() {
+  $entity_field_manager = \Drupal::service('entity_field.manager');
+  $fields = [];
+  // Go through all our text-based field types,
+  // get their field information, and bundle 'em.
+  foreach (['string', 'string_long', 'text_long', 'text_with_summary', 'link'] as $type) {
+    $new_fields = $entity_field_manager->getFieldMapByFieldType($type);
+    foreach ($new_fields as $bundle => $bundle_fields) {
+      $fields[$bundle] = isset($fields[$bundle]) ? array_merge($fields[$bundle], $bundle_fields) : $bundle_fields;
+    }
+  }
+  return $fields;
+}
+
+/**
+ * Search all fields for a needle.
+ *
+ * @param array $fields
+ *   The field names and info to be searched.
+ * @param string $needle
+ *   The string for which to search.
+ * @param bool $regexed
+ *   Whether a regex condition should be used over base LIKE.
+ *
+ * @return array
+ *   The search results.
+ */
+function search_fields(array $fields, string $needle, bool $regexed = FALSE) {
+  $tables = [];
+  $results = [];
+  foreach ($fields as $entity_type_id => $field_map) {
+    // Tables and columns may be constructed slightly differently
+    // depending on the field type. Additionally, how they are
+    // connected to a parent node may be different.
+    switch ($entity_type_id) {
+      case 'block_content':
+      case 'paragraph':
+        foreach ($field_map as $name => $info) {
+          if (str_starts_with($name, 'field')) {
+            $tables[$entity_type_id . '__' . $name] = [
+              'values' => value_columns($name, $info['type']),
+              'parent' => ['entity_id'],
+              'type' => $entity_type_id,
+              'field' => $name,
+            ];
+          }
+        }
+        break;
+
+      case 'node':
+        foreach ($field_map as $name => $info) {
+          if (str_starts_with($name, 'field')) {
+            $tables['node__' . $name] = [
+              'values' => value_columns($name, $info['type']),
+              'parent' => ['entity_id'],
+              'type' => $entity_type_id,
+              'field' => $name,
+            ];
+          }
+          if ($name == 'title') {
+            $tables['node_field_data'] = [
+              'values' => ['title'],
+              'parent' => ['nid'],
+              'type' => $entity_type_id,
+              'field' => $name,
+            ];
+          }
+          if ($name == 'body') {
+            $tables['node__body'] = [
+              'values' => ['body_value'],
+              'parent' => ['entity_id'],
+              'type' => $entity_type_id,
+              'field' => $name,
+            ];
+          }
+        }
+        break;
+    }
```

Comment:
I think this could be simplified to:

```php
$parent_id = 'entity_id';

foreach ($field_map as $name => $info) {
  $table_name = $entity_type_id . '__' . $name;
  $values = value_columns($name, $info['type']);

  switch ($entity_type_id) {
    case 'node':
      $table_name = 'node__' . $name;
      if ($name === 'title') {
        $table_name = 'node_field_data';
        $values = ['title'];
        $parent_id = 'nid';
      }
      elseif ($name === 'body') {
        $value = 'body_value';
      }
      
      break;

  }
  $tables[$table_name] [
    'values' => $values,
    'parent' => [$parent_id],              
    'type' => $entity_type_id,
    'field' => $name,
  ];
}
```